### PR TITLE
add you.com integration

### PR DESCRIPTION
**Pull request by sam-h-bean**
# Add you.com as a search engine option

Before this PR, only Google was used as a search engine. With this PR, the search functionality gets generlized making it possible to choose between Google & [You.com](https://you.com/)

Screenshot 2023-10-18 at 11 21 18

## How to make You.com the selected search engine for chat-ui
1. Get your you API key `YDC_API_KEY` ([docs.you.com](https://docs.you.com))
2. Put it in your `.env.local` file of chat-ui

Screenshot of you.com being used as search engine for chat-ui
image

### CODE REVIEW

**Comment by julien-c** [comment]
[internal] some context [in our internal Slack](https://huggingface.slack.com/archives/C05TF34D6GK/p1697491176563039?thread_ts=1697483269.984629&cid=C05TF34D6GK)

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "chat-ui",
-	"version": "0.6.0",
+	"version": "0.7.0",
```
Comment (gary149) [inline package.json:3]: Not sure we want to bump the version just for this, wdyt @nsarrazin ?

File: README.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ PUBLIC_APP_DISCLAIMER=
 
 ### Web Search config
 
-You can enable the web search by adding either `SERPER_API_KEY` ([serper.dev](https://serper.dev/)) or `SERPAPI_KEY` ([serpapi.com](https://serpapi.com/)) to your `.env.local`.
+You can enable the web search by adding any of `YDC_API_KEY` ([docs.you.com](https://docs.you.com)) or `SERPER_API_KEY` ([serper.dev](https://serper.dev/)) or `SERPAPI_KEY` ([serpapi.com](https://serpapi.com/)) to your `.env.local`.
```
Comment (gary149) [inline README.md:125]: Curious, the only way to get an API key is by sending them an email?

Comment (gary149) [reply README.md:125]: Maybe start the sentence with the SERPER_API_KEY option then?

**Review by gary149 — COMMENTED** [review]
Leaved a few comments but LGTM 👍

File: README.md (hunk 2 above)
Comment (sam-h-bean) [reply README.md:125]: Yes at least until we have a full self serve option up and running

**Comment by mishig25** [comment]
@sam-h-bean created a PR against your branch https://github.com/sam-h-bean/chat-ui/pull/4 on your fork

**Comment by sam-h-bean** [comment]
> @sam-h-bean created a PR against your branch [sam-h-bean#4](https://github.com/sam-h-bean/chat-ui/pull/4) on your fork

merged

File: package.json (hunk 1 above)
Comment (sam-h-bean) [reply package.json]: downgraded!

**Comment by davidli3100** [comment]
Think this would still need some work to bypass the web scraping since we automatically provide the text snippets.
Maybe those snippets just need to be chunked?

**Comment by mishig25** [comment]
@sam-h-bean @davidli3100 https://github.com/sam-h-bean/chat-ui/pull/5

**Review by mishig25 — APPROVED** [review]
lgtm!

**Comment by mishig25** [comment]
@nsarrazin could you check the CI error?

**Comment by sam-h-bean** [comment]
> @nsarrazin could you check the CI error?

taken care of!

**Comment by julien-c** [comment]
this is quite cool, we might switch the provider on our main instance hf.co/chat to _you_! 😁

We should coordinate on some comms if we do make this switch, WDYT?

File: .env [hunk 3]
```diff
@@ -10,6 +10,7 @@ HF_ACCESS_TOKEN=#hf_<token> from from https://huggingface.co/settings/token
 HF_API_ROOT=https://api-inference.huggingface.co/models
 
 # used to activate search with web functionality. disabled if none are defined. choose one of the following:
+YDC_API_KEY=#your serper.dev api key here
```
Comment (mishig25) [inline .env:13]:
```suggestion
YDC_API_KEY=#your docs.you.com api key here
```

**Comment by sam-h-bean** [comment]
> We should coordinate on some comms if we do make this switch, WDYT

Love this! ❤️

**Comment by mishig25** [comment]
Here we made a excel sheets comparing google vs you.com results:

<a href="https://docs.google.com/spreadsheets/d/1pNoDa00d-yoOQt5SwVczDeOO8atsbY3YTK002FwUIPU/edit#gid=1326141439">

</a>

File: src/lib/server/websearch/searchWeb.ts [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+}
+
 // Show result as JSON
 export async function searchWeb(query: string) {
+	if (YDC_API_KEY) {
+		return await searchWebYouApi(query);
+	}
 	if (SERPER_API_KEY) {
 		return await searchWebSerper(query);
 	}
```
Comment (sam-h-bean) [inline src/lib/server/websearch/searchWeb.ts:19]:
```suggestion
	if (SERPER_API_KEY) {
		return await searchWebSerper(query);
	}
	if (YDC_API_KEY) {
		return await searchWebYouApi(query);
	}
```

**Review by nsarrazin — APPROVED** [review]
PR looks good to me! Thanks for the contribution we'll be merging this now.

**Pull request merged**